src/Microsoft.AspNet.Authentication/RemoteAuthenticationOptions.cs

```diff
@@ -50,7 +50,7 @@ public string DisplayName
 
         /// <summary>
         /// Defines whether access and refresh tokens should be stored in the
-        /// <see cref="ClaimsPrincipal"/> after a successful authentication.
+        /// <see cref="ClaimsPrincipal"/> after a successful authorization with the remote provider.
```

Comment:
While our model allows dynamic setting of redirecturi it does not allow for dynamically accepting a uri. Every IDP requires registering a redirecturi, is there a place where the MW can examine the 'path' a message arrived and decide if it should proceed?